scripts/prebuild/mdx-transforms/exclude-content/ast-utils.mjs

```diff
@@ -100,60 +100,151 @@ export function parseDirectiveBlocks(tree) {
  * @param {number} endLine End line (inclusive)
  */
 export function removeNodesInRange(tree, startLine, endLine) {
-	function removeFromNodes(nodes) {
+	function removeFromNodes(nodes, depth = 0, parentInsideRange = false) {
 		if (!Array.isArray(nodes)) {
-			return
+			return parentInsideRange
 		}
 
 		const indicesToRemove = []
-		let insideRange = false
+		let insideRange = parentInsideRange
+
+		const debug = false // Set to true for debugging: process.env.DEBUG_AST_REMOVAL === 'true'
+		const indent = '  '.repeat(depth)
 
 		for (let i = 0; i < nodes.length; i++) {
 			const node = nodes[i]
 			const hasPosition = node.position?.start?.line && node.position?.end?.line
 
+			// Recursively check children FIRST to find any nested END comments
+			// before deciding whether to remove this node
+			// const hadChildren = !!node.children
+			const wasInsideRange = insideRange
+			if (node.children) {
```

Comment:
Thanks for adding tests to cover all of this. It definitely seemed like this rabbit hole went deep. 😢